components/function-controller/internal/git/fetcher.go

```diff
@@ -9,19 +11,42 @@ type git2goFetcher struct {
 }
 
 func (g *git2goFetcher) git2goFetch(url, outputPath string, remoteCallbacks git2go.RemoteCallbacks) (*git2go.Repository, error) {
-	repo, err := git2go.InitRepository(outputPath, true)
+	repo, err := g.openInitRepo(outputPath)
 	if err != nil {
-		return nil, errors.Wrap(err, "while initializing repository")
+		return nil, errors.Wrap(err, "while initializing/opening repository")
 	}
 
-	remote, err := repo.Remotes.Create("origin", url)
+	remote, err := g.lookupCreateRemote(repo, url, outputPath)
 	if err != nil {
-		return nil, errors.Wrap(err, "while creating remote")
+		return nil, errors.Wrap(err, "while creating/using remote")
 	}
 	defer remote.Free()
+
 	err = remote.Fetch(nil, &git2go.FetchOptions{RemoteCallbacks: remoteCallbacks}, "")
 	if err != nil {
 		return nil, errors.Wrap(err, "while fetching remote")
 	}
 	return repo, nil
 }
+
+func (g *git2goFetcher) openInitRepo(outputPath string) (*git2go.Repository, error) {
+	var repo *git2go.Repository
+	var err error
+	repo, err = git2go.OpenRepository(outputPath)
+	if err == nil {
+		return repo, nil
+	}
+	// TODO: this should be in debug
+	fmt.Errorf("failed to open existing repo at [%s]: %v", outputPath, err)
```

Comment:
Any chance we could address it in this PR?
_Same question for:_ [lookup](https://github.com/kyma-project/kyma/pull/14614/files#diff-9ac9090f2d1ba4901158d479850061bb5171ee883d87700476e00c932eaf953cR49-R50)